File: test/test_transform_reduce.cpp
#include <stdio.h>
#include "TRTCContext.h"
#include "DVVector.h"
#include "transform_reduce.h"

int main()
{

	Functor absolute_value = { {},{ "x" }, "        return x<(decltype(x))0 ? -x : x;\n" };

	int h_data[6] = { -1, 0, -2, -2, 1, -3 };
	DVVector d_data("int32_t", 6, h_data);

	ViewBuf res;
	TRTC_Transform_Reduce(d_data, absolute_value, DVInt32(0), Functor("Maximum"), res);
	printf("%d\n", *(int*)res.data());


	return 0;
}

File: CSharp/test/test_transform_reduce/test_transform_reduce.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ThrustRTCSharp;

namespace test_transform_reduce
{
    class test_transform_reduce
    {
        static void Main(string[] args)
        {
            Functor absolute_value = new Functor ( new string[]{ "x" }, "        return x<(decltype(x))0 ? -x : x;\n" );
            DVVector d_data = new DVVector(new int[] { -1, 0, -2, -2, 1, -3 });
            Console.WriteLine(TRTC.Transform_Reduce(d_data, absolute_value, new DVInt32(0), new Functor("Maximum")));
        }
    }
}
